### DISCUSSION TITLE
How to customize/fork nova?

### DISCUSSION THREAD

**Discussion by nrueckmann** [laravel/nova-issues#4152] [Q&A] [answered]
I do not want to argue about the decision to close the private repository. 

But i would like to know if there is any process documented/planned on how to customize/fork Nova.
I know you guys are doing your best to make every customer happy, but some customers (we included) are using Nova production environments and might not be able to wait for a fix or change from your side.

**Comment by milewski** [comment, marked as answer]
- Copy the repository to your own git repository (in nova 3 there was a download button here https://nova.laravel.com/releases but in nova 4 it redirects to the documentation (can we please bring back the download button?) so you will have to download it via composer first and copy the directory to your own repository.. repeat the process for every update)

- Push it to a branch-like `v4`

- Update your composer.json to depend on `"laravel/nova": "v4.x-dev"`

- Add your own repository settings to your composer.json

```php
"repositories": [
    {
        "type": "vcs",
        "url": "git@github.com:your-own-organization/laravel-nova.git"
    }
],
```

- Run composer update

Done, now you are in control of the source code

**Reply by nrueckmann** [reply]
Thank you for your response. I was hoping that there is some kind of official supported workflow. 
In the meanwhile i stick to my custom workflow which includes rebasing my changes on a clean nova branch.

**Reply by RhysLees** [reply]
Make sure your make it a private git repo or you will breach tos https://nova.laravel.com/terms
